### ISSUE TITLE
Gnome no longer offering to integrate Appimage since 3.0.20

### ISSUE THREAD

**Issue by phipolis** [standardnotes/desktop#430]
[Gnome 3.32.2 on Fedora 30] 

With every version through 3.0.19, I would download the -x86_64.AppImage build and mark it executable. Then launching it in the Gnome file browser would display a dialog asking if I want to integrate the new version with the system. Answering yes would cause the app to show up in system App lists.

Starting with 3.0.20, Standard Notes opens without displaying the integrate with system prompt. This is inconvenient, because it no longer shows up in the system launcher.

I can't find a code change in 3.0.20 that seems likely to be responsible. I assume it was a change to the building/packaging process. I notice the -x86_64 is dropped from the file name starting with the same release.

**Comment by affe42** [comment]
Same problem with PureOS 9 64bit (Gnome 3.30.2). Started with SN 3.0.20.

**Comment by aaantoszek** [comment]
Happens also on Elementary OS 5.0 Juno.

**Comment by aysunrhn** [comment]
I'm having the same issue here with Ubuntu 18.04 as well

**Comment by nekojet** [comment]
I saw this same issue yesterday when installing Standard-Notes-3.0.20.AppImage in a new instance of Fedora 31 KDE Plasma spin. Today, I did some testing and found the following...

- executing standard-notes-3.0.19-x86_64.AppImage (and previous versions) triggers an AppImageKit(?) dialog offering to integrate the AppImage with the system, which basically creates ~/.local/share/applications/appimagekit-standard-notes.desktop, which gets picked up by the DE's menu system

- executing Standard-Notes-3.0.21.AppImage does not trigger a dialog, so no option is presented that will create the .desktop file in ~/.local/share/applications/

This behavior is identical on the following:
  Debian 10.1 (buster) x86_64, Xfce 4.12
  Ubuntu 18.04.3 LTS x86_64, GNOME 3.28.x
  Linux Mint 19.2 Tina x86_64, Cinnamon 4.2.x
  Fedora release 30 (Thirty) x86_64, KDE Plasma 5.15.5
  Fedora release 31 (Thirty One) x86_64, KDE Plasma 5.16.5

Hope this helps!

**Comment by mobitar** [comment]
Seems that this is no longer provided by electron-builder?

https://github.com/linuxdeepin/developer-center/issues/837#issuecomment-473643606

https://www.electron.build/configuration/appimage

<img width="703" alt="Screen Shot 2019-11-14 at 12 30 18 PM" src="https://user-images.githubusercontent.com/3277844/68885312-89a2b380-06da-11ea-8965-51f20749161b.png">

**Comment by nekojet** [comment]
Well, that would do it. 😏 It looks like it's up to end users to either:

- install appimaged to run in the background
  https://github.com/AppImage/appimaged

- install and use AppImageLauncher
  https://github.com/TheAssassin/AppImageLauncher

- manage our own .desktop files for AppImages

or just launch AppImages without desktop integration. For now I'm going to manage my own .desktop files while I look into/test the other options.

Thanks for your help. 👍

**Comment by KINGCorgi** [comment]
Would this happen explain why I've had difficult directing the app to my home / application folder when using the terminal? (Excuse my lack of knowledge, I'm new to Ubuntu/Linux). As OP mentioned, for some reason the application doesn't show up in the Application List. 

((I apologize if this isn't related))

**Comment by nekojet** [comment]
AppImageLauncher may do what we're/you're looking for.  I just did a quick test on Ubuntu 18.04 LTS (Bionic Beaver).  I downloaded the latest version of AppImageLauncher (v2.0.3) from here:

https://github.com/TheAssassin/AppImageLauncher/releases

In my case, the version I want is:

appimagelauncher_2.0.3-travis888.4f7bc8e.bionic_amd64.deb

(it may be different depending on which version of Ubuntu/Linux you are running)

After installing AppImageLauncher (right-click the .deb file, Open With Software Install, Install), when I double-clicked on:

Standard-Notes-3.0.21.AppImage

AppImageLauncher asked if I wanted to integrate it. I confirmed that I did, and AppImageLauncher did the following:

- moved Standard-Notes-3.0.21.AppImage to ~/Applications/Standard-Notes-3.0.21_da18992bee2967665af8f6b8283f0bc8.AppImage
- marked Standard-Notes-3.0.21_da18992bee2967665af8f6b8283f0bc8.AppImage as executable
- created  ~/.local/share/applications/appimagekit_45aac0652f59b337ea960eb13266c43b-Standard_Notes.desktop

GNOME sees the .desktop file and adds Standard Notes to its menu system.  Pretty slick.

Hope this helps.

**Comment by phipolis** [comment]
@mobitar is correct, the issue is outside of Standard Notes. And as @nekojet carefully summarized, there are options for getting the original behavior back. That solves it for me, feel free to close!

**Comment by andrewrothman** [comment]
Just installed AppImageLauncher and I was able to get the Standard Notes shortcut back in KDE. Thanks!

**Comment by aaantoszek** [comment]
Yup, the Launcher solves this issue

**Comment by rendhalver** [comment]
I have to ask. What's wrong with Debian or RPM packages?

**Comment by andrewrothman** [comment]
Debian and RPM packages are great, but AppImage is also a very useful format to support. You can run applications packaged as an AppImage without installing it, and without root privileges (which you need in order to install as a Debian package, and I think RPM works this way as well).

**Comment by rendhalver** [comment]
Yes you need root to install packages. It's been that way on most (if not all) systems for a very long time.
If that's the only thing wrong with them then everything else has been doing it wrong since 1970.
AppImage and snap and flatpak's are nice (when they work) but people still use Debian and RPM packages and still like to install things with them. Your base system (usually) uses one of those formats already so why stop there?
Another great thing about deb and rpm is your supporting libraries are usually already installed so you don't need to shoehorn half an operating system into your package to distribute it so it's going to be smaller and will likely run better because it's using those already installed shared libraries.

**Comment by andrewrothman** [comment]
@rendhalver All very valid points. It'd be really nice to have official support for all three, so the user can choose what works best for their use-case. AppImages are also portable, so they can be moved onto a flash drive and carried around if the user would like.

**Comment by rendhalver** [comment]
Atom is built with electron and builds debian and rpm packages I believe.
It's probably as easy as turning it on. :)
I will have a go at working out how to do that and put in a PR.

**Comment by rendhalver** [comment]
#457
